2022/incoming/fractal-christmas-tree.pod

```diff
@@ -0,0 +1,959 @@
+        width  => 1024,                     # hard-coded 1024x768 resolution
+        height => 768,
+        depth  => 32,                       # hard-coded 32-bit color
+        resizeable => 1                     # allow window resize; does not scale window contents
+    );
+    my $my_SDL_event = SDL::Event->new;
+
+    # [ RENDER STATIC GRAPHICS ]
+
+    # draw Christmas tree branches & snow tinsel & ornaments & lights;
+    # iterate through triangle groups in reverse order, due to reverse population during recursion
+    for (my $i = ((scalar @{$my_triangle_groups}) - 1); $i >= 0; $i--) {
+        my number::arrayref::arrayref::arrayref $my_triangle_group = $my_triangle_groups->[$i];
+        my integer::arrayref $my_color = $my_triangle_colors->[$i];
+
+        for (my $j = 0; $j < (scalar @{$my_triangle_group}); $j++) {
+            my number::arrayref::arrayref $my_triangle = $my_triangle_group->[$j];
+
+            # https://metacpan.org/dist/SDL/view/lib/pods/SDLx/Surface.pod
+            $my_SDL_app->draw_trigon_filled( $my_triangle, $my_color );
+
+            # refresh window on every triangle for fun cascade drawing effect
+            $my_SDL_app->update();
+        }
+    }
+
+    # draw Christmas tree trunk & Star of Bethlehem
+    $my_SDL_app->draw_rect( $my_rectangle_trunk, $my_color_brown );
+    $my_SDL_app->draw_trigon_filled( $my_triangle_star_up, $my_color_yellow );
+    $my_SDL_app->draw_trigon_filled( $my_triangle_star_down, $my_color_yellow );
+    $my_SDL_app->update();  # refresh window
+
+    # [ RENDER DYNAMIC (ANIMATED) GRAPHICS ]
+
+    # set initial index for accesssing Christmas tree lights colors
+    my integer $my_lights_colors_index = 0;
+
+    # set initial time for changing Christmas tree lights colors
+    (my integer $seconds_start) = gettimeofday();
+    #print 'have $seconds_start = ', $seconds_start, "\n";
+
+    # the main run loop, used instead of calling $my_SDL_app->run();
+    # animate forever, until SDL_QUIT event is received in GUI window via <Alt-F4> keypress or window close mouse click,
+    # or in CLI window via <Ctrl-C> keypress
+    while(1)
+    {
+        # pump the event loop, gathering events from input devices
+        SDL::Events::pump_events();
+
+        # poll for currently pending events
+        if(SDL::Events::poll_event($my_SDL_event))
+        {
+            print 'have @SDL_EVENTS[', $my_SDL_event->type(), '] = ', @SDL_EVENTS[$my_SDL_event->type()], "\n";
+
+            # we only care about the SDL_QUIT event telling us to exit
+            if ($my_SDL_event->type == SDL_QUIT) {
+                print 'SDL_QUIT event received, exiting', "\n";
+                exit;
+            }
+        }
+ 
+        # get current time, for comparison with start time of current Christmas tree lights color
+        (my integer $seconds_current) = gettimeofday();
+    #    print 'have $seconds_current = ', $seconds_current, "\n";
```

Comment:
The formatting is a bit off here. Was this supposed to be commented out?

Reply:
Yes the print line is supposed to be commented out.
It can optionally be uncommented if anybody wants to see the value of `$seconds_current`.